master/internal/command/command.go

```diff
@@ -510,21 +292,54 @@ func (c *command) setPriority(priority int, forward bool) error {
 	return nil
 }
 
-func (c *command) stringID() string {
-	return c.taskID.String()
+// DeleteIfInWorkspace deletes a command's allocation matching a workspaceID.
+func (c *Command) DeleteIfInWorkspace(req *apiv1.DeleteWorkspaceRequest) {
```

Comment:
style nit, makes a lot more sense to have the caller call ToV1Command().WorkspaceID and KillNTSC, and for them to impl this logic client side than adding this very specific API to Command that it is unlikely for someone else to have a use for.